packages/core/src/Table/TableCell.js

```diff
@@ -1,69 +1,129 @@
 import propTypes from '@dhis2/prop-types'
 import cx from 'classnames'
-import React from 'react'
-import css from 'styled-jsx/css'
-
-const tableCellStyles = css`
-    td {
-        border-bottom: 1px solid #e8edf2;
-        font-size: 14px;
-        line-height: 18px;
-        padding: 13px 12px;
-        height: 45px;
-    }
-
-    .dense {
-        padding: 9px 12px;
-        height: 36px;
-    }
-`
+import React, { forwardRef } from 'react'
+import styles from './TableCell/TableCell.styles.js'
 
 /**
  * @module
  * @param {TableCell.PropTypes} props
  * @returns {React.Component}
- * @example import { TableCell } from '@dhis2/ui-core'
- * @see Live demo: {@link /demo/?path=/story/table--static-layout|Storybook}
+ * @example import { TableCell } from '@dhis2/ui'
+ * @see Live demo: {@link /demo/?path=/story/table--default|Storybook}
  */
-export const TableCell = ({
-    className,
-    children,
-    colSpan,
-    rowSpan,
-    dense,
-    dataTest,
-}) => (
-    <td
-        colSpan={colSpan}
-        rowSpan={rowSpan}
-        className={cx({ dense }, className)}
-        data-test={dataTest}
-    >
-        {children}
-
-        <style jsx>{tableCellStyles}</style>
-    </td>
+export const TableCell = forwardRef(
+    (
+        {
+            active,
+            align,
+            bordered,
+            children,
+            className,
+            colSpan,
+            dataTest,
+            error,
+            fixed,
+            large,
+            left,
+            muted,
+            rowSpan,
+            scope,
+            tag,
+            valid,
+            width,
+            onClick,
+        },
+        ref
+    ) => {
+        const CellTag = tag || fixed ? 'th' : 'td'
```

Comment:
I'm not sure how this is interpreted, but it might need:
```js
tag || (fixed ? 'th' : 'td')
```

Reply:
The expression on the right hand side of the `||` will not be evaluated if the part on the left is truthy, so we're OK ([reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR#short-circuit_evaluation)).